### PR TITLE
doc: mgmt: mcumgr: Add list of libraries/tools

### PR DESCRIPTION
**Pull request by nordicjm**
Adds a list of tools and libraries that are available

### CODE REVIEW

**Comment by nordicjm** [comment]
@kartben do we have better tick/cross symbols to use in doc pages?

**Comment by kartben** [comment]
> @kartben do we have better tick/cross symbols to use in doc pages?

✓ / ✕ or ✔ / ✖, please. 
A bit concerned about the super wide table (definitely broken on the PDF output) so if you can find smart ways to present the information differently, that would be great. I wouldn't consider this blocking though, and worst case scenario we could even exclude the section from the PDF. One thing that comes to mind would be removing iOS and Android column for now (still far too wide, but it helps :)).

File: doc/services/device_mgmt/mcumgr.rst
Comment (kartben) [inline doc/services/device_mgmt/mcumgr.rst:1]: while it's definitely not possible and completely ok for the table to be >100 char, other doc content should wrap at 100.

**Comment by kartben** [comment]
> > @kartben do we have better tick/cross symbols to use in doc pages?
> 
> ✓ / ✕ or ✔ / ✖, please. 

dark theme issue with the later, eh? Unicode is so much fun.

**Comment by nordicjm** [comment]
> > > @kartben do we have better tick/cross symbols to use in doc pages?
> > 
> > 
> > ✓ / ✕ or ✔ / ✖, please.
> 
> dark theme issue with the later, eh? Unicode is so much fun.

Isn't it just! :smile:

**Review by kartben — COMMENTED** [review]
So this basically looks good to me (see one comment re: in-tree client), I am simply not sure what to do with the PDF. Since I don't think that even with the best typesetting engine we could find this table would ever look good or be automatically reformatted to fit in A4 constraints, I think we are looking at either removing the table from the PDF altogether (`.. only:: html`), or there should be another variation of this information that suits a page output better (in a `.. only:: latex` block). For the latter, it wouldn't be terrible to just have a bullet list with all the various links, and maybe adding "(license, app type)" ..?

I know this is extra work but, again, I don't think there can be a magic solution in this particular example.

**Comment by nordicjm** [comment]
@kartben is there a way it can be forced to be rotated for the PDF so it is landscape instead of portrait? Then it should fit I would think

File: doc/services/device_mgmt/mcumgr.rst [hunk 1]
```diff
@@ -33,6 +33,42 @@ the Zephyr tree.
...
+    | and `iOS                                                                       |         |       |     |        |          |        |           |     |    |     |      |          |    |       |        |               | Swift      |         |
+    | <https://github.com/NordicSemiconductor/IOS-nRF-Connect-Device-Manager>`_      |         |       |     |        |          |        |           |     |    |     |      |          |    |       |        |               |            |         |
+    +--------------------------------------------------------------------------------+---------+-------+-----+--------+----------+--------+-----------+-----+----+-----+------+----------+----+-------+--------+---------------+------------+---------+
+    | Zephyr MCUmgr client (in-tree)                                                 | ✕       | ✓     | ✕   | ✕      | ✓        | ✓      | ✓         | ✓   | ✓  | ✓   | ✕    | ✕        | ✕  | ✕     | ✕      | Library       | C          | Apache  |
```
[the remark this answers is not in the document]
Comment (nordicjm) [reply doc/services/device_mgmt/mcumgr.rst:66]: Unfortunately not, I thought I asked the submitter to add documentation but seemingly I didn't, so no docs, just the files and unit tests

**Comment by kartben** [comment]
> @kartben is there a way it can be forced to be rotated for the PDF so it is landscape instead of portrait? Then it should fit I would think
> 
> Had a look and found this https://stackoverflow.com/questions/50620759/sphinx-doc-generate-a-landscape-page-for-long-table-or-graph so seems possible with an extension?

Ya I was about to point to the exact same thread! Seems pretty straightforward, let me give it a try (assuming you might not want to test building the PDF locally :))

File: doc/services/device_mgmt/mcumgr.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (kartben) [reply doc/services/device_mgmt/mcumgr.rst:66]: then the below would probably be quite helpful to make it easier to locate it. Full table to save you the hassle

```

    +---------------------------------------------------------------------------------------------+-------------------------------------------+--------------------------+--------------------------------------------------+---------------+------------+---------+
    | Name                                                                                        | OS support                                | Transports               | Groups                                           | Type          | Language   | License |
    |                                                                                             +---------+-------+-----+--------+----------+--------+-----------+-----+----+-----+------+----------+----+-------+--------+               |            |         |
    |                                                                                             | Windows | Linux | mac | Mobile | Embedded | Serial | Bluetooth | UDP | OS | IMG | Stat | Settings | FS | Shell | Zephyr |               |            |         |
    +=============================================================================================+=========+=======+=====+========+==========+========+===========+=====+====+=====+======+==========+====+=======+========+===============+============+=========+
    | `AuTerm <https://github.com/thedjnK/AuTerm/>`_                                              | ✓       | ✓     | ✓   | ✕      | ✕        | ✓      | ✓         | ✓   | ✓  | ✓   | ✓    | ✓        | ✓  | ✓     | ✓      | Application   | C++ (Qt)   | GPLv3   |
    +---------------------------------------------------------------------------------------------+---------+-------+-----+--------+----------+--------+-----------+-----+----+-----+------+----------+----+-------+--------+---------------+------------+---------+
    | `mcumgr-client <https://github.com/vouch-opensource/mcumgr-client/>`_                       | ✓       | ✓     | ✓   | ✕      | ✕        | ✓      | ✕         | ✕   | ✕  | ✓   | ✕    | ✕        | ✕  | ✕     | ✕      | Application   | Rust       | BSD     |
    +---------------------------------------------------------------------------------------------+---------+-------+-----+--------+----------+--------+-----------+-----+----+-----+------+----------+----+-------+--------+---------------+------------+---------+
    | `mcumgr-web <https://github.com/boogie/mcumgr-web/>`_                                       | ✓       | ✓     | ✓   | ✕      | ✕        | ✕      | ✓         | ✕   | ✕  | ✓   | ✕    | ✕        | ✕  | ✕     | ✕      | Web page      | Javascript | MIT     |
    |                                                                                             |         |       |     |        |          |        |           |     |    |     |      |          |    |       |        | (chrome only) |            |         |
    +---------------------------------------------------------------------------------------------+---------+-------+-----+--------+----------+--------+-----------+-----+----+-----+------+----------+----+-------+--------+---------------+------------+---------+
    | nRF Connect Device Manager: |br|                                                            |         |       |     |        |          |        |           |     |    |     |      |          |    |       |        |               |            |         |
    | `Android                                                                                    | ✕       | ✕     | ✕   | ✓      | ✕        | ✕      | ✓         | ✕   | ✓  | ✓   | ✓    | ✓        | ✓  | ✓     | ✓      | Library and   | Java,      | Apache  |
    | <https://github.com/NordicSemiconductor/Android-nRF-Connect-Device-Manager/>`_              |         |       |     |        |          |        |           |     |    |     |      |          |    |       |        | application   | Kotlin,    |         |
    | and `iOS                                                                                    |         |       |     |        |          |        |           |     |    |     |      |          |    |       |        |               | Swift      |         |
    | <https://github.com/NordicSemiconductor/IOS-nRF-Connect-Device-Manager>`_                   |         |       |     |        |          |        |           |     |    |     |      |          |    |       |        |               |            |         |
    +---------------------------------------------------------------------------------------------+---------+-------+-----+--------+----------+--------+-----------+-----+----+-----+------+----------+----+-------+--------+---------------+------------+---------+
    | Zephyr MCUmgr client (:zephyr_file:`in-tree <subsys/mgmt/mcumgr/smp_client/src/client.c>`)  | ✕       | ✓     | ✕   | ✕      | ✓        | ✓      | ✓         | ✓   | ✓  | ✓   | ✕    | ✕        | ✕  | ✕     | ✕      | Library       | C          | Apache  |
    +---------------------------------------------------------------------------------------------+---------+-------+-----+--------+----------+--------+-----------+-----+----+-----+------+----------+----+-------+--------+---------------+------------+---------+
```

[the remark this answers is not in the document]
Comment (nordicjm) [reply doc/services/device_mgmt/mcumgr.rst:66]: That file is not overly useful, that's just SMP, there's also groups (with separate Kconfigs). MCUmgr client support really needs to be documented @jheiskan81

[the remark this answers is not in the document]
Comment (kartben) [reply doc/services/device_mgmt/mcumgr.rst:66]: kkk

**Comment by kartben** [comment]
@nordicjm  close enough but... meh

image

**Comment by nordicjm** [comment]
That is very disappointing

**Comment by nordicjm** [comment]
@kartben see if this helps with the pdf version

**Comment by de-nordic** [comment]
Shouldn't there also be the mcumgr-cli GO(ne decrepit) client?

**Comment by nordicjm** [comment]
> Shouldn't there also be the mcumgr-cli GO(ne decrepit) client?

That would need it's own category:
User-trash-full

**Comment by de-nordic** [comment]
Maybe the array should be divided into two:
1) OS supported, transport , license
2) groups

**Comment by kartben** [comment]
@nordicjm I didn't want to push force to your branch as I've add bad experiences with it before, but see the two added commits, you'll want to make the pdflscape one first as part of your rebase. Thanks!

There are still some stupid layout decisions made by LaTeX it would seem, but this is good enough I'd say.

image

**Comment by nordicjm** [comment]
Made some additional minifications to put things on multiple lines, hopefully this works slightly better

**Comment by kartben** [comment]
> Made some additional minifications to put things on multiple lines, hopefully this works slightly better

I'll check but I don't think `br` does anything in LaTeX output (probably should... I will try to look into it), hence why I only left it at adding white spaces to the X/Y/Z "words" to try and have them break a bit more easily

**Review by de-nordic — APPROVED** [review]

**Comment by nordicjm** [comment]
@kartben did the break work in the PDF?

**Comment by nordicjm** [comment]
Let's go with it as-is, and see if we can maybe improve it at a later date

**Review by kartben — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**